lib/ansible/modules/storage/purestorage/purefa_volume.py

```diff
@@ -217,7 +226,13 @@ def update_volume(module, array):
         vol_qos['bandwidth_limit'] = 0
     if module.params['size']:
         if human_to_bytes(module.params['size']) != vol['size']:
```

Comment:
This is under the volume size update block, this means if there is a qos limit, would only be able to remove IF you change the volume size.  (if I'm reading right..)
```python
if human_to_bytes(module.params['size']) != vol['size']:
```
I think it should be an elif to the below if:
```python
if human_to_bytes(module.params['qos']) != vol_qos['bandwidth_limit']:
            if 549755813888 >= int(human_to_bytes(module.params['qos'])) >= 1048576:	            if 549755813888 >= int(human_to_bytes(module.params['qos'])) >= 1048576
```

Reply:
Oops - yep put it in the wrong section - well caught #embarrassed